backend/hdf5/h5x/H5Group.cpp

```diff
@@ -217,8 +218,12 @@ DataSet H5Group::createData(const std::string &name,
         HErr res = H5Pset_chunk(dcpl.h5id(), rank, chunks.data());
         res.check("Could not set chunk size on data set creation plist");
     }
-
-    DataSet ds = H5Dcreate(hid, name.c_str(), fileType.h5id(), space.h5id(), H5P_DEFAULT, dcpl.h5id(), H5P_DEFAULT);
+    DataSet ds;
+    if (compression == Compression::DeflateNormal) {
+         HErr status = H5Pset_deflate (dcpl.h5id(), 6);
+         status.check("Could not set compression!");
+    }
```

Comment:
Hmm, maybe it would be good to already add the case where an unsupported compression is specified (i.e. throw an exception)?

Reply:
I think that is exactly what ``check``does, As I see it, maybe I am wrong, ``HErr::check`` throws an [H5Error](https://github.com/G-Node/nix/blob/f352378c222c16d24e01f5c1f79398fb52fd1151/backend/hdf5/h5x/H5Object.hpp#L199)

Reply:
or would you prefer an extra exception for just this case?

Reply:
What I meant is that, if we add a Compression::XXX in the future, we would just ignore it in that function here, right?